File: movieApp/app/src/main/java/com/lbc/practice/movieapp/view/main/MainContract.java
package com.lbc.practice.movieapp.view.main;

import com.lbc.practice.movieapp.adapter.PlayingListContract;
import com.lbc.practice.movieapp.model.PlayingMovieResult;

public interface MainContract {

    interface MainView {
        void moveToFavorite();
        void moveToSecond(PlayingMovieResult.ResultsBean resultsBean);
        void errorMassage(String error);
        void increasePageCount();
    }

    interface MainPre {
        void setPlayingAdaterModel(PlayingListContract.PlayingModel model);
        void setPlayingAdaterView(PlayingListContract.View view);
        void movieInfo(String apiKey,String language, int page);
    }
}

File: movieApp/app/src/main/java/com/lbc/practice/movieapp/view/main/MainContract.kt
package com.lbc.practice.movieapp.view.main;

import com.lbc.practice.movieapp.adapter.PlayingListContract;
import com.lbc.practice.movieapp.data.PlayingMovieResult;

interface MainContract {

    interface MainView {
        fun moveToFavorite()
        fun moveToSecond(resultsBean: PlayingMovieResult.ResultsBean)
        fun errorMassage(error: String)
        fun increasePageCount()
    }

    interface MainPre {
        fun setPlayingAdaterModel(model: PlayingListContract.PlayingModel)
        fun setPlayingAdaterView(view: PlayingListContract.View)
        fun movieInfo(apiKey: String, language: String, page: Int)
    }
}
